lib/elastic_apm/spies/elasticsearch.rb

```diff
@@ -32,10 +32,21 @@ def install
 
           def perform_request(method, path, *args, &block)
             name = format(NAME_FORMAT, method, path)
-            statement = args[0].is_a?(String) ? args[0] : args[0].to_json
+            statement = []
+
+            statement << { params: args&.[](0) }
+
+            if ElasticAPM.agent.config.capture_elasticsearch_queries
+              @sanitizer = ElasticAPM::Transport::Filters::HashSanitizer.new
+              unless args[1].nil? || args[1].empty?
+                statement << {
+                  body: @sanitizer.strip_from!(args[1])
+                }
+              end
```

Comment:
We can't set instance vars inside patched objects – worst case we override and break something 😅

```suggestion
              unless args[1].nil? || args[1].empty?
                body = 
                  ElasticAPM::Transport::Filters::HashSanitizer.new
                  .strip_from!(args[1])
                statement << { body: body }
              end
```

We should probably save the sanitizer in `ElasticAPM::Spies::Elasticsearch` instead so as to not allocate a new one for each and every ES query.

Reply:
https://github.com/mikker/apm-agent-ruby/blob/master/lib/elastic_apm/spies/sequel.rb#L12-L14

Reply:
@mikker right! I followed the sequel pattern and saved the sanitizer in the Spy now, much better :smile: